dagger/e2e.go

```diff
@@ -38,17 +38,39 @@ func e2e(
 		WithExec([]string{"/replicated", "cluster", "create", "--distribution", distribution, "--version", version, "--wait", "15m", "--output", "json"})
 
 	out, err := ctr.Stdout(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to create cluster: %w", err)
-	}
 
+	// Try to unmarshal cluster info even on error, as the cluster may have been partially created
```

Comment:
In retrospect this is incorrect, the parsing logic should remain where it is now